pkg/server/api/entry/v1/service.go

```diff
@@ -466,6 +476,14 @@ func fieldsFromEntryProto(ctx context.Context, proto *types.Entry, inputMask *ty
 		fields[telemetry.TTL] = proto.Ttl
 	}
 
+	if inputMask == nil || inputMask.X509SvidTtl {
```

Comment:
This is a tough one, I think. I've enumerated what I think are all of the cases with values set or unset, during an upgrade or downgrade. There are 42 total situations, of which 9 have potential issues where an unexpected value is used for the SVID TTL.

I'm not sure if there is a set of alternative approaches we can implement which will solve **ALL** the potential issues while also not introducing others. That said, I'm keen for any and all suggestions because you are correct - there are concerns here.

---

1. `TTL` is unset, `X509SvidTTL` is unset, `JWTSvidTTL` is unset
   1. When executing in this version
      1. X509-SVIDs: **No Issue**, generated with the config or global default value.
      2. JWT-SVIDs: **No Issue**, generated with hardcoded `5m` value
   2. When downgrading to Previous Version
      1. X509-SVIDs: **No Issue**, generated with the config or global default value.
      2. JWT-SVIDs: **No Issue**, generated with hardcoded `5m` value
   3. When updating which deletes `TTL` field
      1. X509-SVID: **No Issue**, generated with the config or global default value.
      2. JWT-SVID: **No Issue**, generated with hardcoded `5m` value

2. `TTL` is set, `X509SvidTTL` is unset, `JWTSvidTTL` is unset
   1. When executing in this version
      1. X509-SVIDs: **No Issue**, generated with `TTL` value
      2. JWT-SVIDs: **No Issue**, generated wit hardcoded `5m` value
   2. When downgrading to Previous Version
      1. X509-SVIDs: **No Issue**, continue to be generated with `TTL` value.
      2. JWT-SVIDs: **No Issue**, continue to be generated with hardcoded `5m` value.
   3. When updating which deletes `TTL` field
      1. X509-SVID: **Issue**, now generated with the config or global default value.
      2. JWT-SVID: **No Issue**, continue to be generated with hardcoded `5m` value.

3. `TTL` is set, `X509SvidTTL` is set, `JWTSvidTTL` is unset
   1. When executing in this version
      1. X509-SVIDs: **No Issue**, generated with `X509SvidTTL` value
      2. JWT-SVIDs: **No Issue**, generated with hardcoded `5m` value
   2. When downgrading to Previous Version
      1. X509-SVIDs: **Issue**, generated with `TTL` value, unexpected
      2. JWT-SVIDs: **No Issue**, generated with hardcoded `5m` value
   3. When updating which deletes `TTL` field
      1. X509-SVID: **No Issue**, generated with `X509SvidTTL` value
      2. JWT-SVID: **No Issue**, generated with hardcoded `5m` value

4. `TTL` is set, `X509SvidTTL` is unset, `JWTSvidTTL` is set
   1. When executing in this version
      1. X509-SVIDs: **No Issue**, generated with `TTL` value
      2. JWT-SVIDs: **No Issue**, generated with `JWTSvidTTL` value
   2. When downgrading to Previous Version
      1. X509-SVIDs: **No Issue**, generated with `TTL` value
      2. JWT-SVIDs: **Issue**, generated with hardcoded `5m` value
   3. When updating which deletes `TTL` field
      1. X509-SVID: **Issue**, generated with the config or global default value
      2. JWT-SVID: **No Issue**, generated with `JWTSvidTTL` value

5. `TTL` is set, `X509SvidTTL` is set, `JWTSvidTTL` is set
   1. When executing in this version
      1. X509-SVIDs: **No Issue**, generated with `X509SvidTTL` value
      2. JWT-SVIDs: **No Issue**, generated with `JWTSvidTTL` value
   2. When downgrading to Previous Version
      1. X509-SVIDs: **Issue**, generated with `TTL` value, unexpected
      2. JWT-SVIDs: **Issue**, generated with hardcoded `5m` value, unexpected
   3. When updating which deletes `TTL` field
      1. X509-SVIDs: **No Issue**, generated with `X509SvidTTL` value
      2. JWT-SVIDs: **No Issue**, generated with `JWTSvidTTL` value

6. `TTL` is unset, `X509SvidTTL` is set, `JWTSvidTTL` is unset
   1. When executing in this version
      1. X509-SVIDs: **No Issue**, generated with `X509SvidTTL` value
      2. JWT-SVIDs: **No Issue**, generated with hardcoded `5m` value
   2. When downgrading to Previous Version
      1. X509-SVIDs: **Issue**, generated with the config or global default value. 
      2. JWT-SVIDs: **No Issue**, generated with hardcoded `5m` value
   3. When updating which deletes `TTL` field
      1. X509-SVID: **No Issue**, generated with `X509SvidTTL` value
      2. JWT-SVID: **No Issue**, generated with hardcoded `5m` value

7. `TTL` is unset, `X509SvidTTL` is set, `JWTSvidTTL` is set
   1. When executing in this version
      1. X509-SVIDs: **No Issue**, generated with `X509SvidTTL` value
      2. JWT-SVIDs: **No Issue**, generated with `JWTSvidTTL` value
   2. When downgrading to Previous Version
      1. X509-SVIDs: **Issue**, generated with the config or global default value. 
      2. JWT-SVIDs: **Issue**, generated with hardcoded `5m` value
   3. When updating which deletes `TTL` field
      1. X509-SVID: **No Issue**, generated with `X509SvidTTL` value
      2. JWT-SVID: **No Issue**, generated with `JWTSvidTTL` value

Issues found in `2.iii.a`, `3.ii.a`, `4.ii.b`, `4.iii.a`, `5.ii.a`, `5.ii.b`, `6.ii.a`, `7.ii.a`, `7.ii.b`

### Proposed Solutions

1. If `TTL` is unset and `X509SvidTTL` is being set, then set `TTL` to the value of `X509SvidTTL`
   * Fixes: `6.ii.a`, `7.ii.a`
   * New Issues: None

2. If `TTL` is set and `X509SvidTTL` is unset, then set `X509SvidTTL` to the value of `TTL`
   * Fixes: `2.iii.a`, `4.iii.a`
   * New Issues: None

3. If `TTL` is set and `JWTSvidTTL` is unset, then set `JWTSvidTTL` to the value of `TTL`
   * Fixes: `5.ii.b`
   * New Issues:
      * `2.i.b`, `2.ii.b`, `2.iii.b`, generated with `TTL` value instead of hardcoded `5m` value.
      * `3.i.b`, `3.ii.b`, `3.iii.b`, generated with `TTL` value instead of hardcoded `5m` value.

Issues `3.ii.a`, `4.ii.b`, `5.ii.a`, `7.ii.b` are not fixed by these proposals. 

I do think it's worthwhile to implement proposals (1) and (2). What do you think?

Reply:
We settled on this by simply repurposing the ttl field as x509-svid-ttl at both the api and storage layer. The api layer also renamed the field to make the use clear to consumers. The database column cannot be renamed without migration pain so will remain untouched. The new and unused x509_svid_ttl column will be dropped in a later release.